### PR TITLE
Add AAD V2 support to aks-preview

### PR DESCRIPTION
**Pull request by karataliu**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by karataliu** [comment]
@andyzhangx  I thought the rule changed:

Saw following in the PR creation template`
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify src/index.json.`

**Comment by yonzhan** [comment]
add to S167

File: src/aks-preview/azext_aks_preview/custom.py [hunk 1]
```diff
@@ -1214,6 +1237,15 @@ def aks_update(cmd,     # pylint: disable=too-many-statements,too-many-branches,
         instance.api_server_access_profile = \
             _populate_api_server_access_profile(api_server_authorized_ip_ranges, instance)
 
+    if update_aad_profile:
+        if instance.aad_profile is None or not instance.aad_profile.managed:
+            raise CLIError('Cannot specify "--update-aad-tenant-id/--update-aad-admin-group-object-ids"'
```
Comment (yungezz) [inline src/aks-preview/azext_aks_preview/custom.py]: remove `update` to align with params names

File: src/aks-preview/setup.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 from codecs import open as open1
 from setuptools import setup, find_packages
 
-VERSION = "0.4.36"
+VERSION = "0.4.37"
```
Comment (yungezz) [inline src/aks-preview/setup.py]: this will be a new release or bundle with others?

File: src/aks-preview/azext_aks_preview/custom.py (hunk 1 above)
Comment (karataliu) [reply src/aks-preview/azext_aks_preview/custom.py]: nice, fixed

File: src/aks-preview/setup.py (hunk 2 above)
Comment (karataliu) [reply src/aks-preview/setup.py]: i think we can ship a new release with this, it's a major feature

**Comment by karataliu** [comment]
@yungezz comment addressed, can you please take a look?

**Comment by yungezz** [comment]
general LGTM, just curious any test done/added for this major feature?

**Comment by karataliu** [comment]
@yungezz  Yes, in server side. I didn't see existing command specific test in this repo. Will add into az-cli repo.

**Comment by karataliu** [comment]
@yungezz @arrownj 
I've no doubt adding test cases,  that's absolutely positive request. 
I have mentioned there is no existing parameter test in this repo, and I've confirmed to be adding that in azure-cli repo.

There are other PRs that adds new parameters, is there any special reason only this PR is to be asked to add test for aks-preview?
https://github.com/Azure/azure-cli-extensions/pull/1263
https://github.com/Azure/azure-cli-extensions/pull/1288
https://github.com/Azure/azure-cli-extensions/pull/1319
https://github.com/Azure/azure-cli-extensions/pull/1364

**Comment by arrownj** [comment]
> @yungezz @arrownj
> I've no doubt adding test cases,  that's absolutely positive request.
> I have mentioned there is no existing parameter test in this repo, and I've confirmed to be adding that in azure-cli repo.
> There are other PRs that adds new parameters, is there any special reason only this PR is to be asked to add test for aks-preview?
> #1263
> #1288
> #1319
> #1364

@karataliu , sorry for the inconsistence between each PR. As you know, there are different engineers which may do the review work, and each of them would have different attentions. We are continuing improving the review process and also we should reach a alignment for some standard. Thanks for your feedback, and I think maybe we can start this test standard from now on.^^Thanks for your understanding.

For your case, I think you can only pay attention to the parameter you added.

**Comment by karataliu** [comment]
@arrownj  it'sok, can you point out the dev guide on how to run test in this repo
```
File "test_aks_commands.py", line 12, in <module>
    from azure.cli.testsdk import (
ModuleNotFoundError: No module named 'azure'
```

**Comment by karataliu** [comment]
@yungezz @arrownj test added, would you please take a look?

**Comment by karataliu** [comment]
ping @yungezz @arrownj

**Comment by arrownj** [comment]
> ping @yungezz @arrownj

> @arrownj  it'sok, can you point out the dev guide on how to run test in this repo
> File "test_aks_commands.py", line 12, in <module>
>     from azure.cli.testsdk import (
> ModuleNotFoundError: No module named 'azure'

Hi @karataliu , you can reference our azdev document(https://github.com/Azure/azure-cli-dev-tools) about how to test.

**Pull request merged**